libs/vm-modules/tests/unit/state_tests.cpp

```diff
@@ -539,4 +540,141 @@ TEST_F(
+      ref_array[0] = 1.0fp32;
+      ref_array[1] = 101.01fp32;
+      ref_array[2] = 10101.0101fp32;
+
+      var state = State<Array<Fixed32>>("my array");
+      state.set(ref_array);
+    endfunction
+  )";
+
+  std::string const state_name{"my array"};
+  EXPECT_CALL(toolkit.observer(), Write(state_name, _, _));
+
+  ASSERT_TRUE(toolkit.Compile(ser_src));
+  ASSERT_TRUE(toolkit.Run());
+
+  static char const *deser_src = R"(
+    function main() : Array<Fixed32>
+      var retrieved_state = State<Array<Fixed32>>("my array");
+      return retrieved_state.get(Array<Fixed32>(0));
+    endfunction
+  )";
+
+  EXPECT_CALL(toolkit.observer(), Exists(state_name));
+  EXPECT_CALL(toolkit.observer(), Read(state_name, _, _));
+
+  ASSERT_TRUE(toolkit.Compile(deser_src));
+
+  Variant output;
+  ASSERT_TRUE(toolkit.Run(&output));
+  ASSERT_FALSE(output.IsPrimitive());
+  auto retval{output.Get<Ptr<IArray>>()};
+  ASSERT_TRUE(static_cast<bool>(retval));
+  ASSERT_EQ(int32_t{3}, retval->Count());
+
+  EXPECT_EQ(fetch::fixed_point::fp32_t{1.0},
+            retval->PopFrontOne().Get<fetch::fixed_point::fp32_t>());
+  EXPECT_EQ(fetch::fixed_point::fp32_t{101.01},
+            retval->PopFrontOne().Get<fetch::fixed_point::fp32_t>());
+  EXPECT_EQ(fetch::fixed_point::fp32_t{10101.0101},
+            retval->PopFrontOne().Get<fetch::fixed_point::fp32_t>());
+}
+
+TEST_F(StateTests, test_serialisation_of_fixed_point2)
+{
+  static char const *ser_src = R"(
+    function main()
+      var ref_array = Array<Fixed64>(3);
+      ref_array[0] = 1.0fp64;
+      ref_array[1] = 101.01fp64;
+      ref_array[2] = 10101.0101fp64;
+
+      var state = State<Array<Fixed64>>("my array");
+      state.set(ref_array);
+    endfunction
+  )";
+
+  std::string const state_name{"my array"};
+  EXPECT_CALL(toolkit.observer(), Write(state_name, _, _));
+
+  ASSERT_TRUE(toolkit.Compile(ser_src));
+  ASSERT_TRUE(toolkit.Run());
+
+  static char const *deser_src = R"(
+    function main() : Array<Fixed64>
+      var retrieved_state = State<Array<Fixed64>>("my array");
+      return retrieved_state.get(Array<Fixed64>(0));
+    endfunction
+  )";
+  EXPECT_CALL(toolkit.observer(), Exists(state_name));
+  EXPECT_CALL(toolkit.observer(), Read(state_name, _, _));
+
+  ASSERT_TRUE(toolkit.Compile(deser_src));
+
+  Variant output;
+  ASSERT_TRUE(toolkit.Run(&output));
+  ASSERT_FALSE(output.IsPrimitive());
+  auto retval{output.Get<Ptr<IArray>>()};
+  ASSERT_TRUE(static_cast<bool>(retval));
+  ASSERT_EQ(int32_t{3}, retval->Count());
+  EXPECT_EQ(fetch::fixed_point::fp64_t{1.0},
+            retval->PopFrontOne().Get<fetch::fixed_point::fp64_t>());
+  EXPECT_EQ(fetch::fixed_point::fp64_t{101.01},
+            retval->PopFrontOne().Get<fetch::fixed_point::fp64_t>());
+  EXPECT_EQ(fetch::fixed_point::fp64_t{10101.0101},
+            retval->PopFrontOne().Get<fetch::fixed_point::fp64_t>());
+}
+
+TEST_F(StateTests, test_serialisation_of_fixed_point3)
+{
```

Comment:
Absolutely not critical, but if we had 
test_serialisation_of_fixed_point_32
test_serialisation_of_fixed_point_64
test_serialisation_of_fixed_point_128
instead of 1,2,3, it would be a bit more descripitve and convenient to use in --gtest_filter=... for partial test running.

Reply:
indeed, fixed